scan/lib/scan/report_collector.rb

```diff
@@ -62,6 +64,8 @@ def determine_output_file_name(type)
         "compile_commands.json"
       elsif !@custom_report_file_name.nil?
         @custom_report_file_name
+      elsif type == 'junit'
+        Dir::Tmpname.make_tmpname('scan_', '.xml')
```

Comment:
Wow, yes that's a very good point. I hadn't considered that. I'll have a think about this overnight and see if there is some way to implement this keeping backward compatibility. If not, I'll close the PR.